test/integration/integration.ts

```diff
@@ -1,124 +1,187 @@
+import {
+  FormattedTrustline,
+  SignedTransaction,
+} from "../../src/common/types/objects";
+import { GetBalances } from "../../src/ledger/balances";
+import { Transaction } from "../../src/models/transactions";
 import { generateXAddress } from "../../src/utils/generateAddress";
 import requests from "../fixtures/requests";
 
 import { payTo, ledgerAccept } from "./utils";
-import wallet from "./wallet";
+import { walletAddress, walletSecret } from "./wallet";
 
 // how long before each test case times out
 const TIMEOUT = 20000;
-const INTERVAL = 1000; // how long to wait between checks for validated ledger
 
+// eslint-disable-next-line node/no-process-env -- Allows the user to pass in different IP's for local rippled
 const HOST = process.env.HOST ?? "0.0.0.0";
+// eslint-disable-next-line node/no-process-env -- Allows the user to pass in different ports for local rippled
 const PORT = process.env.PORT ?? "6006";
 const serverUrl = `ws://${HOST}:${PORT}`;
 
 console.log(serverUrl);
 
-function acceptLedger(client) {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- The return type is legitimately any
+async function acceptLedger(client: Client): Promise<any> {
   return client.connection.request({ command: "ledger_accept" });
 }
 
-function verifyTransaction(testcase, hash, type, options, txData, account) {
+// eslint-disable-next-line max-params -- Required for running checks while we move through the test
+async function verifyTransaction(
+  testcase: Mocha.Context,
+  hash: string,
+  type: string,
+  options: { minLedgerVersion: number; maxLedgerVersion?: number },
+  account: string
+): Promise<{ txJSON: string }> {
   console.log("VERIFY...");
-  return testcase.client
-    .request({
-      command: "tx",
-      transaction: hash,
-      min_ledger: options.minLedgerVersion,
-      max_ledger: options.maxLedgerVersion,
-    })
-    .then((data) => {
-      assert(data && data.result);
-      assert.strictEqual(data.result.TransactionType, type);
-      assert.strictEqual(data.result.Account, account);
-      assert.strictEqual(data.result.meta.TransactionResult, "tesSUCCESS");
-      if (testcase.transactions != null) {
-        testcase.transactions.push(hash);
-      }
-      return { txJSON: JSON.stringify(txData), id: hash, tx: data };
-    })
-    .catch((error) => {
-      if (error instanceof errors.PendingLedgerVersionError) {
-        console.log("NOT VALIDATED YET...");
-        return new Promise((resolve, reject) => {
-          setTimeout(
-            () =>
-              verifyTransaction(
-                testcase,
-                hash,
-                type,
-                options,
-                txData,
-                account
-              ).then(resolve, reject),
-            INTERVAL
-          );
-        });
-      }
-      console.log(error.stack);
-      assert(false, `Transaction not successful: ${error.message}`);
-    });
+  const client: Client = testcase.client;
```

Comment:
The added linter rules I sent you should make this not necessary.
The goal with linting is to make code nicer, not to make it a pain to satisfy the linter. This type of solution is excessive and annoying, and honestly makes the code a little less readable, so we want to avoid doing things like this.